robyn/router.py

```diff
@@ -19,11 +20,18 @@
 _logger = logging.getLogger(__name__)
 
 
+def lower_http_method(method: HttpMethod):
+    return (str(method))[11:].lower()
+
+
 class Route(NamedTuple):
     route_type: HttpMethod
     route: str
     function: FunctionInfo
     is_const: bool
+    auth_required: bool
+    openapi_name: str
+    openapi_tags: List[str]
```

Comment:
could you share some thoughts behind this implementation?

Reply:
I hope my earlier comment https://github.com/sparckles/Robyn/pull/1066#discussion_r1866869341 explains most of this.

In Robyn.add_route we were allowing a None value for openapi_tags. Rather than deal with the Union all the way through I remove the possibility of None for easier working with types.

You can see Robyn.start already called `Robyn._add_openapi_routes` that now calls `Router.prepare_routes_openapi` for the Robyn.router The prepare then goes through every direct route and every route in all the included subrouters to call `add_openapi_path_obj`

Reply:
I also want to add that this is on the way to making everything that is part of add_route and include_routes much easier to unit_test. The calls to `add_openapi_path_obj` had become side effects when adding a route and so harder to unit test.